samples/embedded-map/src/components/EmbeddedMap/EmbeddedMapModel.ts

```diff
@@ -77,23 +79,87 @@ export default class EmbeddedMapModel extends ComponentModelBase {
 
         // A new instance is being set - add the event handlers.
         if (instance) {
-            // eslint-disable-next-line @typescript-eslint/no-floating-promises
-            (async () => {
-                // Wait for the initial sync to be setup before listening for
-                // events from Mapillary.
-                await this._syncMaps();
-
-                // Track changes when the user moves to a different node, or when
-                // their bearing/tilt position changes.
-                this.mapillary.on(
-                    Viewer.nodechanged,
-                    this._onPerspectiveChange
-                );
-                this.mapillary.on(Viewer.povchanged, this._onPerspectiveChange);
-            })();
+            // Track changes when the user moves to a different node, or when
+            // their bearing/tilt position changes.
+            this.mapillary.on(Viewer.nodechanged, this._onPerspectiveChange);
+            this.mapillary.on(Viewer.povchanged, this._onPerspectiveChange);
+
+            // Change the mapillary viewer position when the location marker is moved
+            this._viewerUpdateHandle = this.messages.events.locationMarker.updated.subscribe(
+                this._handleViewerUpdate.bind(this)
+            );
+        }
+    }
+
+    private _map: GcxMap | undefined;
+    get map(): GcxMap | undefined {
+        return this._map;
+    }
+    @importModel("map-extension")
+    set map(instance: GcxMap | undefined) {
+        if (instance === this._map) {
+            return;
+        }
+
+        // If an instance already exists, clean it up first.
+        if (this._map) {
+            void this._unsyncMaps();
+        }
+
+        this._map = instance;
+
+        // A new instance is being set - sync the map.
+        if (instance) {
+            this.messages.events.map.initialized.subscribe(
+                this._syncMaps.bind(this)
+            );
+
+            if (!document.body.onmousedown) {
+                document.body.onmousedown = () => delete this._lastMarkerUpdate;
```

Comment:
We should use `addEventListener` instead. This is quite brittle.

We also have a leak as they're not being cleaned up.

Did you maybe mean to add an event listener to the map itself instead?

Reply:
I'd prefer not to do this at all, but just listen to an event that only fires with the final position of the marker. I'll definitely clean this up, but this probably shouldn't go in the SDK sample as we could remove the need with a simple fix in Web

Reply:
Right that makes sense. @eric-parsons thoughts?

Reply:
Yes, I agree. Let's get the improvements we discussed in place in Web so that this can get cleaned up. Those extra events and operations will be useful in general beyond this sample.